mozetl/utils.py

```diff
@@ -48,20 +48,15 @@ def write_csv(dataframe, path, header=True):
     is ~15GB for ec2 c3.xlarge (due to caching overhead).
     """
 
-    # NOTE: Before spark 2.1, toLocalIterator will timeout on some dataframes
-    # because rdd materialization can take a long time. Instead of using
-    # an iterator over all partitions, collect everything into driver memory.
-    logger.info("Writing {} rows to {}".format(dataframe.count(), path))
+    folder = "{}.temp".format(path)
+    logger.info("Writing single partition to {}".format(folder))
+    dataframe.repartition(1).write.csv("file://" + folder, header=header)
```

Comment:
I wanted to prevent version specific versions of the `csv.writer` from affecting the results with respect to unicode values. Using the spark dataframe writer avoids the python standard library and uses Spark's internal representation.